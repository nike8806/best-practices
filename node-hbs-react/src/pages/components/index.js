export { default as WorkEmail } from './WorkEmail/WorkEmail.js';
