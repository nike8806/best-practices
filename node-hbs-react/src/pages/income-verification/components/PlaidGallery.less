.plaid-gallery__bank-logo:hover,
.plaid-gallery__find-bank:hover,
.plaid-gallery__find-bank a:hover {
  background-color: #f9f9f9;
  color: #707884;
  cursor: pointer;
  text-decoration: none;
}

.plaid-gallery__bank-logo {
  background-color: #ffffff;
  border: 1px solid #e2e2e2;
  border-radius: 5px;
  display: block;
  margin: 2px;
  background-position: center;
  background-repeat: no-repeat;
  background-size: 65%;
  height: 115px;
}

.plaid-gallery__find-bank {
  background-color: #ffffff;
  border: 1px solid #e2e2e2;
  border-radius: 5px;
}

.plaid-gallery__bank-logo--chase {
  background-image: url('../images/chase-logo.svg');
}

.plaid-gallery__bank-logo--wells {
  background-image: url('../images/wells-fargo-logo.svg');
}

.plaid-gallery__bank-logo--bofa {
  background-image: url('../images/bofa-logo.svg');
}

.plaid-gallery__bank-logo--pnc {
  background-image: url('../images/pnc-logo.svg');
}

.plaid-gallery__bank-logo--us {
  background-image: url('../images/us-bank-logo.svg');
}

.plaid-gallery__bank-logo--usaa {
  background-image: url('../images/usaa-logo.svg');
  background-size: 50%;
}

@media (min-width: 376px) and (max-width:767px) {
  .plaid-gallery__bank-logo {
      background-size: 40%;
      height: 120px;
  }

  .plaid-gallery__bank-logo--usaa {
      background-size: 26%;
  }
}

@media (min-width: 768px) {
  .plaid-gallery__bank-logo {
      background-size: 55%;
  }

  .plaid-gallery__bank-logo--usaa {
      background-size: 40%;
  }
}
