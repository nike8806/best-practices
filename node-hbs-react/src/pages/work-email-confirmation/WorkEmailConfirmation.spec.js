import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import WorkEmailConfirmation from './WorkEmailConfirmation';
import axios from 'lc-axios';
import { SuccessMessage, Modal, Input} from 'components';

// TODO: Put this configuration in just one place
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

const ERROR_SELECTOR = '.work-email-confirmation__modal-error',
  EMAIL_SENT_SELECTOR = '.work-email-confirmation__modal-email-sent',
  MODAL_NOT_WANT_SELECTOR = '.work-email-confirmation__modal-not-want',
  EMAIL_SECTION_SELECTOR = '.work-email-confirmation__email-section';

/**
 * Tested Component reference along the tests cases.
 */
describe('WorkEmail Confirmation Spec React component', () => {
  let component;
  beforeEach(() => {
    component = shallow(<WorkEmailConfirmation />);
  });

  it('WorkEmailConfirmation render correctly', () => {
    expect(component).toMatchSnapshot();
  });

  describe('WorkEmail input', () => {
    let workEmailInput;
    beforeEach(() => {
      workEmailInput = component.find(Input).first();
    });
    describe('On Blur', () => {
      it('The input should have a touched status', () => {
        workEmailInput.simulate('blur', {
          target: {
            value: 'assa@someEmail.com'
          }});
        expect(component.instance().state.touchedWorkEmail).toBe(true);
      });
    });
    describe('On Change', () => {
      it('The input should Not Have an error if the input has the valid format', () => {
        workEmailInput.simulate('change', {
          target: {
            value: 'some@email.tld'
          }
        });
        component.update();
        expect(component.instance().state.errorWorkEmail).toBe(undefined);
      });
      it('The input should have an error "WorkEmail Required" if the value is empty', () => {
        workEmailInput.simulate('change', {
          target: {
            value: 'some@email.tld'
          }
        });
        component.update();
        workEmailInput.simulate('change', {
          target: {
            value: ''
          }
        });
        component.update();
        expect(component.instance().state.errorWorkEmail).toBe('Work email is required.');
      });

      it('The input should have an error if the email is invalid', () => {
        workEmailInput.simulate('change', {
          target: {
            value: 'some@'
          }
        });
        component.update();
        expect(component.instance().state.errorWorkEmail).toBe('Please specify a valid email address');
      });

      it('The input should have an error if the domain is a Provider not permited', () => {
        const errorMessage = `Sorry, we can\'t accept addresses from popular email providers
        (gmail, hotmail, yahoo, etc.). If you don\'t have any other work email address,
        please select \'I don\'t have a work email.\'`;
        workEmailInput.simulate('change', {
          target: {
            value: 'some@gmail.com'
          }
        });
        component.update();
        expect(component.instance().state.errorWorkEmail).toBe(errorMessage);
      });
    });
  });

  describe('On click on submit button', () => {
    let submitButton, workEmailInput;
    beforeEach(() => {
      submitButton = component.update().find('button.work-email-confirmation__btn-submit-email').first();
      workEmailInput = component.update().find(Input).first();
    });
    it('should not send the info if workEmail has an error', () => {
      workEmailInput.simulate('change', {
        target: {
          value: ''
        }
      });
      component.update();
      submitButton.simulate('click');
      expect(component.instance().state.errorWorkEmail).toEqual('Work email is required.');
      expect(component.update().find(EMAIL_SECTION_SELECTOR).length).toBe(1);
      expect(component.find(SuccessMessage).length).toBe(0);
      expect(component.find(Modal).length).toBe(0);
      expect(component.find(EMAIL_SENT_SELECTOR).length).toBe(0);
      expect(component.find(ERROR_SELECTOR).length).toBe(0);
    });
    it('should handle submit making an AJAX success request', (done) => {
      axios.post = jest.fn(() => Promise.resolve({ data: 'foo' }));
      workEmailInput.simulate('change', {
        target: {
          value: 'some@email.tld'
        }
      });
      submitButton.simulate('click');
      Promise.all([axios.post]).then(() => {
        expect(axios.post).toHaveBeenCalledWith('/todo/work-email-confirmation/send', { email: 'some@email.tld' });
        expect(component.update().find(Modal).length).toBe(1);
        expect(component.find(EMAIL_SENT_SELECTOR).length).toBe(1);
        expect(component.find(ERROR_SELECTOR).length).toBe(0);
        expect(component).toMatchSnapshot();
        done();
      });
    });
    it('should show a modal with error if request fails', (done) => {
      axios.post = jest.fn(() => Promise.reject({ error: 'foo' }));
      workEmailInput.simulate('change', {
        target: {
          value: 'some@email.tld'
        }
      });
      submitButton.simulate('click');
      Promise.all([axios.post]).then(() => {
        expect(axios.post).toHaveBeenCalledWith('/todo/work-email-confirmation/send', { email: 'some@email.tld' });
        expect(component.update().find(Modal).length).toBe(1);
        expect(component.find(ERROR_SELECTOR).length).toBe(1);
        expect(component).toMatchSnapshot();
        done();
      });
    });
  });

  describe('On optOut submit', () => {
    describe('Do not have click', () => {
      let doNotHaveButton;
      beforeEach(() => {
        doNotHaveButton = component.find('button.work-email-confirmation__btn-optout-do-not-have').first();
      });
      it('should handle opt out when request response is ok', (done) => {
        axios.post = jest.fn(() => Promise.resolve({ data: 'foo' }));
        doNotHaveButton.simulate('click');
        Promise.all([axios.post]).then(() => {
          expect(axios.post).toHaveBeenCalledWith('/todo/work-email-confirmation/optout', { reason: 'DO_NOT_HAVE' });
          expect(component.update().find(SuccessMessage).length).toBe(1);
          expect(component.update().find(ERROR_SELECTOR).length).toBe(0);
          expect(component).toMatchSnapshot();
          done();
        });
      });
      it('should show a modal with error if request fails', (done) => {
        axios.post = jest.fn(() => Promise.reject({ error: 'foo' }));
        doNotHaveButton.simulate('click');
        Promise.all([axios.post]).then(() => {
          expect(axios.post).toHaveBeenCalledWith('/todo/work-email-confirmation/optout', { reason: 'DO_NOT_HAVE' });
          expect(component.update().find(Modal).length).toBe(1);
          expect(component.find(ERROR_SELECTOR).length).toBe(1);
          expect(component).toMatchSnapshot();
          done();
        });
      });
    });

    describe('Do not want click', () => {
      let openModalDoNotWantButton;
      beforeEach(() => {
        openModalDoNotWantButton = component.find('button.work-email-confirmation__btn-show-opt-out-section').first();
        openModalDoNotWantButton.simulate('click');
        expect(component).toMatchSnapshot();
      });
      it('should show modal with "Do not want" options', () => {
        expect(component.update().find(Modal).length).toBe(1);
        expect(component.find(MODAL_NOT_WANT_SELECTOR).length).toBe(1);
      });

      describe('On "I understand, but I\'d prefer not to" button click', () => {
        let doNotWantButton;
        beforeEach(() => {
          doNotWantButton = component.find('button.work-email-confirmation__btn-optout-do-not-want').first();
        });
        it('should show a success message when get a success response', (done) => {
          axios.post = jest.fn(() => Promise.resolve({ data: 'foo' }));
          doNotWantButton.simulate('click');
          Promise.all([axios.post]).then(() => {
            expect(axios.post).toHaveBeenCalledWith('/todo/work-email-confirmation/optout', { reason: 'DO_NOT_WANT' });
            expect(component.update().find(SuccessMessage).length).toBe(1);
            done();
          });
        });

        it('should show a modal with error if request fails', (done) => {
          axios.post = jest.fn(() => Promise.reject({ error: 'foo' }));
          doNotWantButton.simulate('click');
          Promise.all([axios.post]).then(() => {
            expect(axios.post).toHaveBeenCalledWith('/todo/work-email-confirmation/optout', { reason: 'DO_NOT_WANT' });
            expect(component.update().find(Modal).length).toBe(1);
            expect(component.find(ERROR_SELECTOR).length).toBe(1);
            done();
          });
        });
      });
      describe('On "Go back to the faster way" buttton click', () => {
        let fasterWayButton;
        beforeEach(() => {
          fasterWayButton = component.find('button.work-email-confirmation__btn-fast-way').first();
        });
        it('should close the modal', () => {
          fasterWayButton.simulate('click');
          expect(component.update().find(Modal).length).toBe(0);
          expect(component.update().find(EMAIL_SECTION_SELECTOR).length).toBe(1);
          expect(component.update().find(SuccessMessage).length).toBe(0);
        });
      });
    });
  });
  describe('At modal oppened', () => {
    beforeEach(() => {
      jest.useFakeTimers();
      component = mount(<WorkEmailConfirmation />);
    });

    it('The modal should be closed when the user click outside', () => {
      component.setState({ showModal: 'ERROR' });
      let modalInstance = component.update().find(Modal);
      expect(modalInstance.length).toBe(1);
      modalInstance.find('.modal-backdrop').first().simulate('click', { preventDefault() {} });
      expect(component.update().find(Modal).length).toBe(0);
    });

    it('An ERROR Modal should be closed after 5 seconds', () => {
      component.setState({ showModal: 'ERROR' });
      expect(component.update().find(Modal).length).toBe(1);
      jest.runAllTimers();
      expect(component.update().find(Modal).length).toBe(0);
    });

    it('An EMAIL_SENT Modal should be closed after 5 seconds', () => {
      component.setState({ showModal: 'EMAIL_SENT' });
      expect(component.update().find(Modal).length).toBe(1);
      jest.runAllTimers();
      expect(component.update().find(Modal).length).toBe(0);
    });
  });
});
