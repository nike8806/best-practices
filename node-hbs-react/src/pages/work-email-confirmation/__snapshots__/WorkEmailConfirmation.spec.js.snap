// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkEmail Confirmation Spec React component On click on submit button should handle submit making an AJAX success request 1`] = `
<div
  className="work-email-confirmation"
>
  <div
    className="todo-body"
  >
    <div
      className="work-email-confirmation__email-section"
    >
      <h1
        className="todo-heading big"
      >
        Quickly and easily verify your employment
      </h1>
      <p
        className="todo-text"
      >
        The fastest way to verify your employment is to provide your work email. This email will only be used for verification purposes.
      </p>
      <label
        className="sr-only"
        htmlFor="work-email-confirmation-input"
      >
        Work email
      </label>
      <Input
        className="work-email-confirmation__work-email-control visible-inline"
        errorMessage={null}
        id="work-email-confirmation-input"
        inputClassName="work-email-confirmation__work-email-input visible-inline"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="enter your work email"
        required={true}
        type="email"
        value="some@email.tld"
      />
      <button
        className="btn btn-todo center-block work-email-confirmation__btn-submit-email margin-bottom-32 margin-top-32"
        onClick={[Function]}
        type="button"
      >
        Submit
      </button>
      <h2
        className="todo-heading"
      >
        What happens next?
      </h2>
      <ol
        className="todo-text"
      >
        <li>
          We'll send you a one-time confirmation email (it might take a few seconds).
        </li>
        <li>
          You click on the button in the email.
        </li>
        <li>
          You're done!
        </li>
      </ol>
      <div
        className="row"
      >
        <div
          className="col-xs-12 col-sm-6"
        >
          <button
            className="btn-todo-link work-email-confirmation__btn-show-opt-out-section visible-block margin-bottom-32"
            onClick={[Function]}
            type="button"
          >
            I'd rather not provide my work email
          </button>
        </div>
        <div
          className="col-xs-12 col-sm-6 text-right-sm"
        >
          <button
            className="btn-todo-link work-email-confirmation__btn-optout-do-not-have"
            onClick={[Function]}
            type="button"
          >
            I don't have a work email
          </button>
        </div>
      </div>
    </div>
    <Modal
      bodyClassName={null}
      className="todo-modal work-email-confirmation__modal-email-sent"
      contentClassName={null}
      dialogClassName="work-email-confirmation__modal-dialog"
      footer={null}
      footerClassName={null}
      header={null}
      headerClassName={null}
      onDismiss={[Function]}
      overlayClassName={null}
      role="dialog"
      tabIndex="-1"
      title={null}
      titleClassName={null}
    >
      <h2
        className="todo-heading text-center padding-all-0"
      >
        Email sent! To complete verification click the link in the email you receive.
      </h2>
    </Modal>
  </div>
</div>
`;

exports[`WorkEmail Confirmation Spec React component On click on submit button should show a modal with error if request fails 1`] = `
<div
  className="work-email-confirmation"
>
  <div
    className="todo-body"
  >
    <div
      className="work-email-confirmation__email-section"
    >
      <h1
        className="todo-heading big"
      >
        Quickly and easily verify your employment
      </h1>
      <p
        className="todo-text"
      >
        The fastest way to verify your employment is to provide your work email. This email will only be used for verification purposes.
      </p>
      <label
        className="sr-only"
        htmlFor="work-email-confirmation-input"
      >
        Work email
      </label>
      <Input
        className="work-email-confirmation__work-email-control visible-inline"
        errorMessage={null}
        id="work-email-confirmation-input"
        inputClassName="work-email-confirmation__work-email-input visible-inline"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="enter your work email"
        required={true}
        type="email"
        value="some@email.tld"
      />
      <button
        className="btn btn-todo center-block work-email-confirmation__btn-submit-email margin-bottom-32 margin-top-32"
        onClick={[Function]}
        type="button"
      >
        Submit
      </button>
      <h2
        className="todo-heading"
      >
        What happens next?
      </h2>
      <ol
        className="todo-text"
      >
        <li>
          We'll send you a one-time confirmation email (it might take a few seconds).
        </li>
        <li>
          You click on the button in the email.
        </li>
        <li>
          You're done!
        </li>
      </ol>
      <div
        className="row"
      >
        <div
          className="col-xs-12 col-sm-6"
        >
          <button
            className="btn-todo-link work-email-confirmation__btn-show-opt-out-section visible-block margin-bottom-32"
            onClick={[Function]}
            type="button"
          >
            I'd rather not provide my work email
          </button>
        </div>
        <div
          className="col-xs-12 col-sm-6 text-right-sm"
        >
          <button
            className="btn-todo-link work-email-confirmation__btn-optout-do-not-have"
            onClick={[Function]}
            type="button"
          >
            I don't have a work email
          </button>
        </div>
      </div>
    </div>
    <Modal
      bodyClassName={null}
      className="todo-modal work-email-confirmation__modal-error"
      contentClassName={null}
      dialogClassName="work-email-confirmation__modal-dialog"
      footer={null}
      footerClassName={null}
      header={null}
      headerClassName={null}
      onDismiss={[Function]}
      overlayClassName={null}
      role="dialog"
      tabIndex="-1"
      title={null}
      titleClassName={null}
    >
      <h2
        className="todo-heading text-center padding-all-0"
      >
        Something went wrong. Please try again later.
      </h2>
    </Modal>
  </div>
</div>
`;

exports[`WorkEmail Confirmation Spec React component On optOut submit Do not have click should handle opt out when request response is ok 1`] = `
<div
  className="work-email-confirmation"
>
  <div
    className="todo-body"
  >
    <SuccessMessage
      className=""
      subtitle="Please be sure to finish the rest of your task list."
      title="No problem. Thanks for letting us know"
      withIcon={false}
    >
      <a
        className="btn btn-todo"
        href="/account/myAccount.action"
      >
        Return to task list
      </a>
    </SuccessMessage>
  </div>
</div>
`;

exports[`WorkEmail Confirmation Spec React component On optOut submit Do not have click should show a modal with error if request fails 1`] = `
<div
  className="work-email-confirmation"
>
  <div
    className="todo-body"
  >
    <div
      className="work-email-confirmation__email-section"
    >
      <h1
        className="todo-heading big"
      >
        Quickly and easily verify your employment
      </h1>
      <p
        className="todo-text"
      >
        The fastest way to verify your employment is to provide your work email. This email will only be used for verification purposes.
      </p>
      <label
        className="sr-only"
        htmlFor="work-email-confirmation-input"
      >
        Work email
      </label>
      <Input
        className="work-email-confirmation__work-email-control visible-inline"
        errorMessage={null}
        id="work-email-confirmation-input"
        inputClassName="work-email-confirmation__work-email-input visible-inline"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="enter your work email"
        required={true}
        type="email"
        value=""
      />
      <button
        className="btn btn-todo center-block work-email-confirmation__btn-submit-email margin-bottom-32 margin-top-32"
        onClick={[Function]}
        type="button"
      >
        Submit
      </button>
      <h2
        className="todo-heading"
      >
        What happens next?
      </h2>
      <ol
        className="todo-text"
      >
        <li>
          We'll send you a one-time confirmation email (it might take a few seconds).
        </li>
        <li>
          You click on the button in the email.
        </li>
        <li>
          You're done!
        </li>
      </ol>
      <div
        className="row"
      >
        <div
          className="col-xs-12 col-sm-6"
        >
          <button
            className="btn-todo-link work-email-confirmation__btn-show-opt-out-section visible-block margin-bottom-32"
            onClick={[Function]}
            type="button"
          >
            I'd rather not provide my work email
          </button>
        </div>
        <div
          className="col-xs-12 col-sm-6 text-right-sm"
        >
          <button
            className="btn-todo-link work-email-confirmation__btn-optout-do-not-have"
            onClick={[Function]}
            type="button"
          >
            I don't have a work email
          </button>
        </div>
      </div>
    </div>
    <Modal
      bodyClassName={null}
      className="todo-modal work-email-confirmation__modal-error"
      contentClassName={null}
      dialogClassName="work-email-confirmation__modal-dialog"
      footer={null}
      footerClassName={null}
      header={null}
      headerClassName={null}
      onDismiss={[Function]}
      overlayClassName={null}
      role="dialog"
      tabIndex="-1"
      title={null}
      titleClassName={null}
    >
      <h2
        className="todo-heading text-center padding-all-0"
      >
        Something went wrong. Please try again later.
      </h2>
    </Modal>
  </div>
</div>
`;

exports[`WorkEmail Confirmation Spec React component On optOut submit Do not want click On "Go back to the faster way" buttton click should close the modal 1`] = `
<div
  className="work-email-confirmation"
>
  <div
    className="todo-body"
  >
    <div
      className="work-email-confirmation__email-section"
    >
      <h1
        className="todo-heading big"
      >
        Quickly and easily verify your employment
      </h1>
      <p
        className="todo-text"
      >
        The fastest way to verify your employment is to provide your work email. This email will only be used for verification purposes.
      </p>
      <label
        className="sr-only"
        htmlFor="work-email-confirmation-input"
      >
        Work email
      </label>
      <Input
        className="work-email-confirmation__work-email-control visible-inline"
        errorMessage={null}
        id="work-email-confirmation-input"
        inputClassName="work-email-confirmation__work-email-input visible-inline"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="enter your work email"
        required={true}
        type="email"
        value=""
      />
      <button
        className="btn btn-todo center-block work-email-confirmation__btn-submit-email margin-bottom-32 margin-top-32"
        onClick={[Function]}
        type="button"
      >
        Submit
      </button>
      <h2
        className="todo-heading"
      >
        What happens next?
      </h2>
      <ol
        className="todo-text"
      >
        <li>
          We'll send you a one-time confirmation email (it might take a few seconds).
        </li>
        <li>
          You click on the button in the email.
        </li>
        <li>
          You're done!
        </li>
      </ol>
      <div
        className="row"
      >
        <div
          className="col-xs-12 col-sm-6"
        >
          <button
            className="btn-todo-link work-email-confirmation__btn-show-opt-out-section visible-block margin-bottom-32"
            onClick={[Function]}
            type="button"
          >
            I'd rather not provide my work email
          </button>
        </div>
        <div
          className="col-xs-12 col-sm-6 text-right-sm"
        >
          <button
            className="btn-todo-link work-email-confirmation__btn-optout-do-not-have"
            onClick={[Function]}
            type="button"
          >
            I don't have a work email
          </button>
        </div>
      </div>
    </div>
    <Modal
      bodyClassName={null}
      className="todo-modal work-email-confirmation__modal-not-want"
      contentClassName={null}
      dialogClassName="work-email-confirmation__modal-dialog"
      footer={null}
      footerClassName={null}
      header={null}
      headerClassName={null}
      onDismiss={[Function]}
      overlayClassName={null}
      role="dialog"
      tabIndex="-1"
      title="Just a heads up..."
      titleClassName="todo-heading"
    >
      <p
        className="todo-text"
      >
        If you decide not to provide us with your work email, it may take us longer to verify your employment. We will use this email address to confirm your employment, but never for marketing purposes.
      </p>
      <button
        className="btn btn-todo center-block work-email-confirmation__btn-fast-way margin-bottom-32"
        onClick={[Function]}
      >
        Go back to the faster way
      </button>
      <button
        className="btn-todo-link center-block work-email-confirmation__btn-optout-do-not-want"
        onClick={[Function]}
      >
        I understand, but I'd prefer not to
      </button>
    </Modal>
  </div>
</div>
`;

exports[`WorkEmail Confirmation Spec React component On optOut submit Do not want click On "I understand, but I'd prefer not to" button click should show a modal with error if request fails 1`] = `
<div
  className="work-email-confirmation"
>
  <div
    className="todo-body"
  >
    <div
      className="work-email-confirmation__email-section"
    >
      <h1
        className="todo-heading big"
      >
        Quickly and easily verify your employment
      </h1>
      <p
        className="todo-text"
      >
        The fastest way to verify your employment is to provide your work email. This email will only be used for verification purposes.
      </p>
      <label
        className="sr-only"
        htmlFor="work-email-confirmation-input"
      >
        Work email
      </label>
      <Input
        className="work-email-confirmation__work-email-control visible-inline"
        errorMessage={null}
        id="work-email-confirmation-input"
        inputClassName="work-email-confirmation__work-email-input visible-inline"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="enter your work email"
        required={true}
        type="email"
        value=""
      />
      <button
        className="btn btn-todo center-block work-email-confirmation__btn-submit-email margin-bottom-32 margin-top-32"
        onClick={[Function]}
        type="button"
      >
        Submit
      </button>
      <h2
        className="todo-heading"
      >
        What happens next?
      </h2>
      <ol
        className="todo-text"
      >
        <li>
          We'll send you a one-time confirmation email (it might take a few seconds).
        </li>
        <li>
          You click on the button in the email.
        </li>
        <li>
          You're done!
        </li>
      </ol>
      <div
        className="row"
      >
        <div
          className="col-xs-12 col-sm-6"
        >
          <button
            className="btn-todo-link work-email-confirmation__btn-show-opt-out-section visible-block margin-bottom-32"
            onClick={[Function]}
            type="button"
          >
            I'd rather not provide my work email
          </button>
        </div>
        <div
          className="col-xs-12 col-sm-6 text-right-sm"
        >
          <button
            className="btn-todo-link work-email-confirmation__btn-optout-do-not-have"
            onClick={[Function]}
            type="button"
          >
            I don't have a work email
          </button>
        </div>
      </div>
    </div>
    <Modal
      bodyClassName={null}
      className="todo-modal work-email-confirmation__modal-not-want"
      contentClassName={null}
      dialogClassName="work-email-confirmation__modal-dialog"
      footer={null}
      footerClassName={null}
      header={null}
      headerClassName={null}
      onDismiss={[Function]}
      overlayClassName={null}
      role="dialog"
      tabIndex="-1"
      title="Just a heads up..."
      titleClassName="todo-heading"
    >
      <p
        className="todo-text"
      >
        If you decide not to provide us with your work email, it may take us longer to verify your employment. We will use this email address to confirm your employment, but never for marketing purposes.
      </p>
      <button
        className="btn btn-todo center-block work-email-confirmation__btn-fast-way margin-bottom-32"
        onClick={[Function]}
      >
        Go back to the faster way
      </button>
      <button
        className="btn-todo-link center-block work-email-confirmation__btn-optout-do-not-want"
        onClick={[Function]}
      >
        I understand, but I'd prefer not to
      </button>
    </Modal>
  </div>
</div>
`;

exports[`WorkEmail Confirmation Spec React component On optOut submit Do not want click On "I understand, but I'd prefer not to" button click should show a success message when get a success response 1`] = `
<div
  className="work-email-confirmation"
>
  <div
    className="todo-body"
  >
    <div
      className="work-email-confirmation__email-section"
    >
      <h1
        className="todo-heading big"
      >
        Quickly and easily verify your employment
      </h1>
      <p
        className="todo-text"
      >
        The fastest way to verify your employment is to provide your work email. This email will only be used for verification purposes.
      </p>
      <label
        className="sr-only"
        htmlFor="work-email-confirmation-input"
      >
        Work email
      </label>
      <Input
        className="work-email-confirmation__work-email-control visible-inline"
        errorMessage={null}
        id="work-email-confirmation-input"
        inputClassName="work-email-confirmation__work-email-input visible-inline"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="enter your work email"
        required={true}
        type="email"
        value=""
      />
      <button
        className="btn btn-todo center-block work-email-confirmation__btn-submit-email margin-bottom-32 margin-top-32"
        onClick={[Function]}
        type="button"
      >
        Submit
      </button>
      <h2
        className="todo-heading"
      >
        What happens next?
      </h2>
      <ol
        className="todo-text"
      >
        <li>
          We'll send you a one-time confirmation email (it might take a few seconds).
        </li>
        <li>
          You click on the button in the email.
        </li>
        <li>
          You're done!
        </li>
      </ol>
      <div
        className="row"
      >
        <div
          className="col-xs-12 col-sm-6"
        >
          <button
            className="btn-todo-link work-email-confirmation__btn-show-opt-out-section visible-block margin-bottom-32"
            onClick={[Function]}
            type="button"
          >
            I'd rather not provide my work email
          </button>
        </div>
        <div
          className="col-xs-12 col-sm-6 text-right-sm"
        >
          <button
            className="btn-todo-link work-email-confirmation__btn-optout-do-not-have"
            onClick={[Function]}
            type="button"
          >
            I don't have a work email
          </button>
        </div>
      </div>
    </div>
    <Modal
      bodyClassName={null}
      className="todo-modal work-email-confirmation__modal-not-want"
      contentClassName={null}
      dialogClassName="work-email-confirmation__modal-dialog"
      footer={null}
      footerClassName={null}
      header={null}
      headerClassName={null}
      onDismiss={[Function]}
      overlayClassName={null}
      role="dialog"
      tabIndex="-1"
      title="Just a heads up..."
      titleClassName="todo-heading"
    >
      <p
        className="todo-text"
      >
        If you decide not to provide us with your work email, it may take us longer to verify your employment. We will use this email address to confirm your employment, but never for marketing purposes.
      </p>
      <button
        className="btn btn-todo center-block work-email-confirmation__btn-fast-way margin-bottom-32"
        onClick={[Function]}
      >
        Go back to the faster way
      </button>
      <button
        className="btn-todo-link center-block work-email-confirmation__btn-optout-do-not-want"
        onClick={[Function]}
      >
        I understand, but I'd prefer not to
      </button>
    </Modal>
  </div>
</div>
`;

exports[`WorkEmail Confirmation Spec React component On optOut submit Do not want click should show modal with "Do not want" options 1`] = `
<div
  className="work-email-confirmation"
>
  <div
    className="todo-body"
  >
    <div
      className="work-email-confirmation__email-section"
    >
      <h1
        className="todo-heading big"
      >
        Quickly and easily verify your employment
      </h1>
      <p
        className="todo-text"
      >
        The fastest way to verify your employment is to provide your work email. This email will only be used for verification purposes.
      </p>
      <label
        className="sr-only"
        htmlFor="work-email-confirmation-input"
      >
        Work email
      </label>
      <Input
        className="work-email-confirmation__work-email-control visible-inline"
        errorMessage={null}
        id="work-email-confirmation-input"
        inputClassName="work-email-confirmation__work-email-input visible-inline"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="enter your work email"
        required={true}
        type="email"
        value=""
      />
      <button
        className="btn btn-todo center-block work-email-confirmation__btn-submit-email margin-bottom-32 margin-top-32"
        onClick={[Function]}
        type="button"
      >
        Submit
      </button>
      <h2
        className="todo-heading"
      >
        What happens next?
      </h2>
      <ol
        className="todo-text"
      >
        <li>
          We'll send you a one-time confirmation email (it might take a few seconds).
        </li>
        <li>
          You click on the button in the email.
        </li>
        <li>
          You're done!
        </li>
      </ol>
      <div
        className="row"
      >
        <div
          className="col-xs-12 col-sm-6"
        >
          <button
            className="btn-todo-link work-email-confirmation__btn-show-opt-out-section visible-block margin-bottom-32"
            onClick={[Function]}
            type="button"
          >
            I'd rather not provide my work email
          </button>
        </div>
        <div
          className="col-xs-12 col-sm-6 text-right-sm"
        >
          <button
            className="btn-todo-link work-email-confirmation__btn-optout-do-not-have"
            onClick={[Function]}
            type="button"
          >
            I don't have a work email
          </button>
        </div>
      </div>
    </div>
    <Modal
      bodyClassName={null}
      className="todo-modal work-email-confirmation__modal-not-want"
      contentClassName={null}
      dialogClassName="work-email-confirmation__modal-dialog"
      footer={null}
      footerClassName={null}
      header={null}
      headerClassName={null}
      onDismiss={[Function]}
      overlayClassName={null}
      role="dialog"
      tabIndex="-1"
      title="Just a heads up..."
      titleClassName="todo-heading"
    >
      <p
        className="todo-text"
      >
        If you decide not to provide us with your work email, it may take us longer to verify your employment. We will use this email address to confirm your employment, but never for marketing purposes.
      </p>
      <button
        className="btn btn-todo center-block work-email-confirmation__btn-fast-way margin-bottom-32"
        onClick={[Function]}
      >
        Go back to the faster way
      </button>
      <button
        className="btn-todo-link center-block work-email-confirmation__btn-optout-do-not-want"
        onClick={[Function]}
      >
        I understand, but I'd prefer not to
      </button>
    </Modal>
  </div>
</div>
`;

exports[`WorkEmail Confirmation Spec React component WorkEmailConfirmation render correctly 1`] = `
<div
  className="work-email-confirmation"
>
  <div
    className="todo-body"
  >
    <div
      className="work-email-confirmation__email-section"
    >
      <h1
        className="todo-heading big"
      >
        Quickly and easily verify your employment
      </h1>
      <p
        className="todo-text"
      >
        The fastest way to verify your employment is to provide your work email. This email will only be used for verification purposes.
      </p>
      <label
        className="sr-only"
        htmlFor="work-email-confirmation-input"
      >
        Work email
      </label>
      <Input
        className="work-email-confirmation__work-email-control visible-inline"
        errorMessage={null}
        id="work-email-confirmation-input"
        inputClassName="work-email-confirmation__work-email-input visible-inline"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="enter your work email"
        required={true}
        type="email"
        value=""
      />
      <button
        className="btn btn-todo center-block work-email-confirmation__btn-submit-email margin-bottom-32 margin-top-32"
        onClick={[Function]}
        type="button"
      >
        Submit
      </button>
      <h2
        className="todo-heading"
      >
        What happens next?
      </h2>
      <ol
        className="todo-text"
      >
        <li>
          We'll send you a one-time confirmation email (it might take a few seconds).
        </li>
        <li>
          You click on the button in the email.
        </li>
        <li>
          You're done!
        </li>
      </ol>
      <div
        className="row"
      >
        <div
          className="col-xs-12 col-sm-6"
        >
          <button
            className="btn-todo-link work-email-confirmation__btn-show-opt-out-section visible-block margin-bottom-32"
            onClick={[Function]}
            type="button"
          >
            I'd rather not provide my work email
          </button>
        </div>
        <div
          className="col-xs-12 col-sm-6 text-right-sm"
        >
          <button
            className="btn-todo-link work-email-confirmation__btn-optout-do-not-have"
            onClick={[Function]}
            type="button"
          >
            I don't have a work email
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
